src/main/java/com/benmorant/game/holycraap/model/entity/Item.java

```diff
@@ -25,4 +26,90 @@
   private int slot;
 
   private double price;
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JsonBackReference
+  private People people;
+
+  public Item() {}
+
+  /** Constructeur surchargé pour Item (moins l'ID, auto-généré. */
+  public Item(String name, int slot, double price, People people) {
+    this.name = name;
+    this.slot = slot;
+    this.price = price;
+    this.people = people;
+  }
+
+  public Integer getId() {
+    return id;
+  }
+
+  public void setId(Integer id) {
+    this.id = id;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public int getSlot() {
+    return slot;
+  }
+
+  public void setSlot(int slot) {
+    this.slot = slot;
+  }
+
+  public double getPrice() {
+    return price;
+  }
+
+  public void setPrice(double price) {
+    this.price = price;
+  }
+
+  public People getPeople() {
+    return people;
+  }
+
+  public void setPeople(People people) {
+    this.people = people;
+  }
+
+  /** Méthode equals pour Item. */
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Item)) {
+      return false;
+    }
+    return id != null && id.equals(((Item) o).getId());
+  }
+
+  @Override
+  public int hashCode() {
+    return 31;
+  }
```

Comment:
This means all items will get the same hash in `HashMap`/`HashSet` _etc_... This will result in `HashMap` to be as inefficient as a `List`, as it will require it to do a table lookup for every search. It would be better to leave the hash code as it is (what you're probably looking for is to use the `id` here but I wouldn't recommend playing with the hashing function unless there is a performance problem. Optimizing is rarely a good idea so early in a project).

https://fr.wikipedia.org/wiki/Table_de_hachage
https://en.wikipedia.org/wiki/Hash_table

Reply:
https://pt.wikipedia.org/wiki/Tabela_de_dispers%C3%A3o

Reply:
Kiddin' . Actualy, this article explains why I used that kind of hashcodes :

https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/

(And I checked a JHispster's issue, where the solution actually was following that advise. 

The truth is : while generating the hashcode with IntelliJ, I had to do so many choices. If I remember that well, it was much easier with Eclipse.

I'll check if it works without the hashcode, and if not, I'll generate a "normal" one .

Reply:
It doesn't work without the hashcode, and it doesn't work either when I generate a normal one.

Reply:
The "normal" hashcode works. But I get an error with the "normal" (i.e generated by IntelliJ) equals() : 

  @Override
    public boolean equals(Object o) {
      if (this == o) {
        return true;
      }
      if (!(o instanceof People)) {
        return false;
      }
      People people = (People) o;
      return currentHp == people.currentHp
          && hpMax == people.hpMax
          && currentMp == people.currentMp
          && mpMax == people.mpMax
          && id.equals(people.id)
          && name.equals(people.name)
          && inventory.equals(people.inventory);
    }

So I used the same equals() I used before.

Reply:
I understood why you overloaded `equals`, all I'm saying is that redefining the `hashCode` to a constant is a bad idea.

This new `hashCode` doesn't make sense either given how you implemented `equals`. You want to hash on `id` only if your comparison just checks for `id`.

Reply:
I now have implemented equals following IntelliJ, I just removed the part about inventory, and it works.